str/fine-mapping/finemap_runner.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+"""
+analysis-runner --dataset "bioheart" --access-level "test" --description "Run finemap" --output-dir "str/finemap" finemap_runner.py
+
+"""
+from cpg_utils.hail_batch import get_batch
+
+def main():
+    b = get_batch()
+    input_file_path = 'gs://cpg-bioheart-test/str/associatr/finemap/example/data'
+    data_in = b.read_input_group(**{
+            'k':input_file_path+ '.k',
+            'ld':input_file_path+ '.ld',
+            'z' : input_file_path+ '.z',
+            'master' : input_file_path+ '.master'})
+
+
+
+    eh_job = b.new_job(name=f'Run finemap')
+    eh_job.image('australia-southeast1-docker.pkg.dev/cpg-common/images-dev/finemap:1.4.2')
+    eh_job.command(
+                    f"""
+                # Create a temporary file
+                temp_file=$(mktemp)
+
+                input_dir=$(ls -d ${{BATCH_TMPDIR}}/inputs/*/)
+
+                # Write the required format to the file
+                echo 'z;ld;snp;config;cred;log;k;n_samples' > $temp_file
+                echo '${{input_dir}}data.z;${{input_dir}}data.ld;${{input_dir}}data.snp;data.config;data.cred;data.log;/data.k;5363' >> $temp_file
```

Comment:
Creating the master file at runtime like this via `echo` is the right thing to do. To get the input group filenames into it you'd want to do something like
```suggestion
                echo "{data_in.z};{data_in.z};{data_in.ld};…etc…;5363" >> $temp_file
```

Reply:
ah okay, thanks! I was messing with batch_tmpdir, but this makes sense and is much cleaner :)

Reply:
Double quotes `"` (rather than single `'` quotes)  is going to be important here BTW because the `{data_in.foo}` Hail things will expand to `$BATCH_TMPDIR}/blah/blah` and that variable will need to be expanded by the shell before finemap sees it.